### PR TITLE
Added display locking doc

### PR DESCRIPTION
**Pull request by vmpstr**
Please take a look.

### CODE REVIEW

File: display-locking/all.md [hunk 1]
```diff
@@ -0,0 +1,544 @@
+# Display Locking
```
Comment (chrishtr) [inline display-locking/all.md]: What do you think of this file vs the one at ../display-locking.md?

Comment (vmpstr) [reply display-locking/all.md]: I don't really want to erase the other file, and I was thinking of breaking this one up into finer grained docs? If you think this doc supercedes the other one, then I'm happy to override it.

Comment (vmpstr) [reply display-locking/all.md]: SGTM

**Comment by vmpstr** [comment]
PTAL

File: display-locking/explainer.md [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+the complexity of the DOM, and the number of changes required, this process can
+take a long time. This, again, causes jank since other update phases do not
+happen while layout and paint is executing.
+
+The output of the layout and paint phase is a sequence of draw commands which,
+when executed, will draw the current visual update.
+
+##### Compositing
+In order to avoid doing repeated work, modern user-agents also employ
+compositing. Compositing phase is a process of splitting up draw commands into
```
Comment (chrishtr) [inline display-locking/explainer.md]: s/Compositing phase/Compositing/

File: display-locking/explainer.md [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+acceleration, this update phase issues GPU commands to the driver to display the
+layers and its contents.
+
+At the end of the presentation phase, the content is finally visible to the
+user.
+
+---
+### Proposal
+
+Display locking proposal is intended to improve the layout and paint, as well as
```
Comment (chrishtr) [inline display-locking/explainer.md]: "The display locking proposal"

Also, it's intended to improve the Script phase as well.

File: display-locking/explainer.md [hunk 4]
```diff
@@ -0,0 +1,544 @@
...
+parts of the compositing update phases. In particular, it aims to add javascript
+APIs to allow the developer to lock an element for display. This means that the
+element paint output (ie the output of the layout and paint phase) will not
```
Comment (chrishtr) [inline display-locking/explainer.md]: element subtree

File: display-locking/explainer.md (hunk 2 above)
Comment (vmpstr) [reply display-locking/explainer.md]: Done.

File: display-locking/explainer.md [hunk 5]
```diff
@@ -0,0 +1,544 @@
...
+the interactions happen with this object, which is associated with the element
+that returned the object.
+
+##### DisplayLock.acquire()
+
+Acquire performs the following steps:
+
+* It returns a promise, which resolves when the acquire process is complete,
+  then
+* It finishes all of the update phases for the current state
```
Comment (chrishtr) [inline display-locking/explainer.md]: What are the "update phases"?

File: display-locking/explainer.md [hunk 6]
```diff
@@ -0,0 +1,544 @@
...
+  screen.
+
+##### DisplayLock.commit() [*bikeshed: release()*]
+
+Commit performs the following steps:
+
+* It returns a promise, which resolves when the element enters the unlock
+  pending phase.
+* It finishes all update phases for the locked element, as described above.
+* It enters unlock pending phase and return the promise. Note that at this
```
Comment (chrishtr) [inline display-locking/explainer.md]: It enters the unlock...

File: display-locking/explainer.md [hunk 7]
```diff
@@ -0,0 +1,544 @@
...
+  point, the element is still locked for visual updates. This allows the
+  callback in the promise resolution to re-lock the lock without changing the
+  visual representation of the element.
+* After the script phase is done and the lock has not been re-acquired, the
+  element is unlocked and enters the element is unlocked state.
+
+---
+### Description
```
Comment (chrishtr) [inline display-locking/explainer.md]: Implementation description?

File: display-locking/explainer.md [hunk 8]
```diff
@@ -0,0 +1,544 @@
...
+implementation:
+
+* Threaded update phase: locked subtrees can be processed on a separate thread.
+* Fixed phase updates: locked subtrees could be processed one update phase at a
+  time. For example, layout finishes first and skips the other phases. On the
+  next frame, paint finishes, and skips the remainder. Etc.
+
+It’s possible that other approach could have better behavior than budgeted
+update phases, but any approach is feasible as long as it implements
+co-operative phase updates.
```
Comment (chrishtr) [inline display-locking/explainer.md]: s/co-operative/non-janking/ ?

File: display-locking/explainer.md [hunk 9]
```diff
@@ -0,0 +1,544 @@
...
+  lock.commit().then(function() { onContentPresented(lock); });
+}
+
+window.onload = function() {
+  window.requestAnimationFrame(presentContent);
+};
+</script>
+```
+
+Similar to the original example, onload we present the content in the
```
Comment (chrishtr) [inline display-locking/explainer.md]: on load

File: display-locking/explainer.md (hunk 5 above)
Comment (vmpstr) [reply display-locking/explainer.md]: In the background section, I say that whatever stages are described there I will refer to as "update phases", but they roughly map to lifecycle updates plus presentation to screen... Should I just use lifecycle updates as the term?

**Comment by vmpstr** [comment]
PTAL

File: display-locking/explainer.md (hunk 5 above)
Comment (chrishtr) [reply display-locking/explainer.md]: No it's good as-is.

File: display-locking/explainer.md [hunk 10]
```diff
@@ -0,0 +1,544 @@
...
+acceleration, this update phase issues GPU commands to the driver to display the
+layers and its contents.
+
+At the end of the presentation phase, the content is finally visible to the
+user.
+
+---
+### Proposal
+
+The display locking proposal is intended to improve the script, layout,  paint,
```
Comment (chrishtr) [inline display-locking/explainer.md]: Nit: extra space after "layout,"

**Pull request merged**